### PR TITLE
Protect AccountUser Name with only allowed symbols

### LINKED ISSUE #207: Limiting the characters of account names
It is necessary to prohibit any spaces in the name, non-printable characters (%*&$^#, etc.)

Allow next symbols:
abcdefghijklmnopqrstuvwxyz0123456789_

### PR DESCRIPTION
**Pull request by andyoknen**

### CODE REVIEW

File: src/pocketdb/consensus/social/User.hpp [hunk 1]
```diff
@@ -189,6 +173,21 @@ namespace PocketConsensus
 
             return Success;
         }
+    
+        virtual ConsensusValidateResult CheckLogin(const UserRef& ptx)
+        {
+            auto name = *ptx->GetPayloadName();
+
+            // Trim spaces
+            if (boost::algorithm::ends_with(name, "%20") || boost::algorithm::starts_with(name, "%20"))
```
Comment (lostystyg) [inline src/pocketdb/consensus/social/User.hpp:168]: Why not using same logic as for `UserConsensus_checkpoint_login_limitation` but with checkpoint? Also i think checking for emptiness is reuired

Comment (tawmaz) [reply src/pocketdb/consensus/social/User.hpp:168]: Isn't there also an issue with other non-visible characters such as %00 being included at the beginning and end of the string?

Comment (tawmaz) [reply src/pocketdb/consensus/social/User.hpp:168]: It appears every character from 0x00 to 0x20 is a control character and should be excluded

Comment (andyoknen) [reply src/pocketdb/consensus/social/User.hpp:168]: ::isalnum should identify any bad character
But I'm not sure about the control characters - from 0x00 to 0x20

Comment (lostystyg) [reply src/pocketdb/consensus/social/User.hpp:168]: It ok because allows only a-z, A-Z and 0-9

**Comment by tawmaz** [comment]
Seems like this must be compared to a older commit on 0.20 branch because a lot of the commits listed should already be present in 0.20?  Maybe a rebase is required?

**Comment by andyoknen** [comment]
> Seems like this must be compared to a older commit on 0.20 branch because a lot of the commits listed should already be present in 0.20? Maybe a rebase is required?

I made a merge from 0.20 to remove conflicts

**Review by lostystyg — APPROVED** [review]

**Review by only1question — APPROVED** [review]

**Review by tawmaz — APPROVED** [review]

**Pull request merged**